org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/ui/util/JavaProjectUtilities.java

```diff
@@ -0,0 +1,117 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```

Comment:
Add Red Hat line for having moved logic from other files into this new file.  If there is room, add the files, otherwise, you can add line comments below.